confgenerator/experimental.go

```diff
@@ -14,3 +22,23 @@ func IsExperimentalFeatureEnabled(receiver string) bool {
 	}
 	return false
 }
+
+func registerExperimentalValidations(v *validator.Validate) {
+	v.RegisterStructValidation(componentValidator, ConfigComponent{})
+}
+
+func componentValidator(sl validator.StructLevel) {
+	comp, ok := sl.Current().Interface().(ConfigComponent)
+	if !ok {
+		return
+	}
+	feature, ok := requiredFeatureForType[comp.Type]
+	if !ok || IsExperimentalFeatureEnabled(feature) {
+		return
+	}
+	sl.ReportError(nil, "", "", "experimental", fmt.Sprintf("Component of type %q cannot be used with the current version of the Ops Agent", comp.Type))
```

Comment:
I'm confused about this message (maybe I'm misunderstanding the code). Wouldn't this show only if the feature is disabled ( via "EXPERIMENTAL_FEATURES" ) and not related to the Ops Agent version ? Maybe something like  `"Component of type %q is disabled in the Ops Agent."`

Reply:
From the customer's point of view, the features enabled via this flag are eventually going to be made GA in a future version -- the error message would go away simply by upgrading the Ops Agent version once the feature is GA.